src/Corax/Queries/AllEntriesMatch.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using Voron;
+using Voron.Data.Containers;
+using Voron.Data.Sets;
+using Voron.Global;
+using Voron.Impl;
+
+namespace Corax.Queries
+{
+    public struct AllEntriesMatch : IQueryMatch
+    {
+        private readonly Transaction _tx;
+        private readonly long _count;
+        private Set.Iterator _entriesPagesIt;
+        private int _offset;
+        private Page _currentPage;
+        private long _entriesContainerId;
+
+        public unsafe AllEntriesMatch(Transaction tx)
+        {
+            _tx = tx;
+            _count = tx.LowLevelTransaction.RootObjects.ReadInt64(IndexWriter.NumberOfEntriesSlice) ?? 0;
+            _entriesContainerId = tx.OpenContainer(IndexWriter.EntriesContainerSlice);
+            _entriesPagesIt = Container.GetAllPagesSet(tx.LowLevelTransaction, _entriesContainerId).Iterate();
+            _offset = 0;
+            _currentPage = new Page(null);
+        }
+
+        public long Count => _count;
+        public QueryCountConfidence Confidence => QueryCountConfidence.High;
+        public unsafe int Fill(Span<long> matches)
+        {
+            while (true)
+            {
+                if (_currentPage.IsValid == false)
+                {
+                    if (_entriesPagesIt.MoveNext() == false)
+                    {
+                        return 0;
+                    }
+
+                    _currentPage = _tx.LowLevelTransaction.GetPage(_entriesPagesIt.Current);
+                    _offset = 0;
+                }
+            
+                var results = 0;
+                while (results + 1 < matches.Length)
+                {
+                    var read = Container.GetEntriesInto(_entriesContainerId, _offset, _currentPage, matches);
+                    if (read == 0)
+                    {
+                        _currentPage = new Page(null);
+                        break;
+                    }
+                    results += read;
+                    _offset += read;
+                }
+                return results;
```

Comment:
I mean this return. It's inside that true loop, so it's escaping method in the first iteration.
Also: shouldn't results be declared outside loops?

Reply:
Great catch, you are correct. Fixed now